sphinx/domains/javascript.py

```diff
@@ -316,47 +313,48 @@ class JavaScriptDomain(Domain):
         'mod':   JSXRefRole(),
     }
     initial_data = {
-        'objects': {},  # fullname -> docname, objtype
-        'modules': {},  # modname  -> docname
+        'objects': {},  # fullname -> docname, node_id, objtype
+        'modules': {},  # modname  -> docname, node_id
     }  # type: Dict[str, Dict[str, Tuple[str, str]]]
 
     @property
-    def objects(self) -> Dict[str, Tuple[str, str]]:
-        return self.data.setdefault('objects', {})  # fullname -> docname, objtype
+    def objects(self) -> Dict[str, Tuple[str, str, str]]:
+        return self.data.setdefault('objects', {})  # fullname -> docname, node_id, objtype
 
-    def note_object(self, fullname: str, objtype: str, location: Any = None) -> None:
+    def note_object(self, fullname: str, objtype: str, node_id: str,
+                    location: Any = None) -> None:
```

Comment:
This method takes an additional argument `node_id`. It would be a breaking change...

Reply:
I thought other APIs in JS domain are also changed. And I don't have good idea to keep compatibility. So I'm going with this implementation.